src/main/java/org/web3j/utils/Account.java

```diff
@@ -38,7 +38,7 @@ public static Boolean isAddress(String address) {
 
     private static Boolean validateChecksumAddress(String address) {
         String formattedAddress = address.replace("0x", "").toLowerCase();
-        String hash = Numeric.toHexString(Hash.sha3(formattedAddress.getBytes()));
+        String hash = Numeric.toHexStringNoPrefix(Hash.sha3(formattedAddress.getBytes()));
         for (int i = 0; i < 40; i++) {
             if (Character.isLetter(address.charAt(i))) {
```

Comment:
Why do you check from the address? This seems to be wrong - address still contains the leading '0x'

Reply:
Sure, I missed that method. In this case you can also remove the `.replace("0x", "")` above, which I assumes is what happens in `toHexStringNoPrefix`?

Reply:
Ah I misunderstood your question. This is actually the correct way to do it. The hash needs to be computed from the lowercase address without a prefix, but the validation of the hash should happen against the original, untouched address.

https://github.com/ethereum/EIPs/issues/55

Reply:
No I don't think that you can remove it - if you remove it it would be used in the sha3 hashing.

Reply:
compare with
https://ethereum.stackexchange.com/questions/1374/how-can-i-check-if-an-ethereum-address-is-valid

especially this line:
`address = address.replace('0x','');`
>

Reply:
Oh you are correct - thank you for pointing that out. The checksum validation should be carried out on the address without the 0x prefix. Fixed that now, thanks!

Reply:
Now you are ussing the lower cased version also in the comparision - thats not right.

Reply:
Slow day for me today isn't it? Fixed that now, thanks.